sway/desktop/layer_shell.c

```diff
@@ -659,13 +659,13 @@ void handle_layer_shell_surface(struct wl_listener *listener, void *data) {
 	sway_layer->output_destroy.notify = handle_output_destroy;
 	wl_signal_add(&output->events.destroy, &sway_layer->output_destroy);
 
-	wl_list_insert(&output->layers[layer_surface->client_pending.layer],
+	wl_list_insert(&output->layers[layer_surface->pending.layer],
 			&sway_layer->link);
 
-	// Temporarily set the layer's current state to client_pending
+	// Temporarily set the layer's current state to pending
 	// So that we can easily arrange it
 	struct wlr_layer_surface_v1_state old_state = layer_surface->current;
-	layer_surface->current = layer_surface->client_pending;
+	layer_surface->current = layer_surface->pending;
```

Comment:
Unrelated, but that's a hack and will stop working once https://github.com/swaywm/wlroots/pull/3151 is merged.

Reply:
Ugh, yeah, that isn't great. We should definitely fix it, maybe by passing a "configuring" layer surface to `arrange_layers`. Also `arrange_layers` shouldn't actually update the positions of the other layers just yet… We should wait for `ack_configure`.

Oh well.